module/zfs/zvol.c

```diff
@@ -339,24 +339,6 @@ zvol_get_stats(objset_t *os, nvlist_t *nv)
 	return (SET_ERROR(error));
 }
 
-static void
-zvol_size_changed(zvol_state_t *zv, uint64_t volsize)
-{
-	struct block_device *bdev;
-
-	ASSERT(MUTEX_HELD(&zv->zv_state_lock));
-
-	bdev = bdget_disk(zv->zv_disk, 0);
-	if (bdev == NULL)
-		return;
-
-	set_capacity(zv->zv_disk, volsize >> 9);
-	zv->zv_volsize = volsize;
-	check_disk_size_change(zv->zv_disk, bdev);
```

Comment:
Is there anything we're losing from no longer using `check_disk_size_change`? 

It looks like there's kernel logging:
`printk(KERN_INFO, "%s: detected capacity change from %lld to %lld\n", disk->disk_name, bdev_size, disk_size);`

and updates to the `inode`: 
`i_size_write(bdev->bd_inode, disk_size);`

I'm not sure if the `inode` is meaningful to us in this case, I don't fully understand how it relates to a`zvol`. I also don't know if kernel logging makes sense as a level of abstraction for a `zvol` but maybe we can get the size change logged in `zfs_dbgmsg`?

Reply:
@shartse good question.  The short answer is we shouldn't be losing anything since `revalidate_disk()` will internally call `check_disk_size_change()`.  In fact, `revalidate_disk()` is the interface we should have been using all along, and `check_disk_size_change()` was un-exported because it wasn't originally intended to be called by drivers.

As for logging the size change that's an interesting thought, what would be the use case for this? 
 In practice my feeling is it wouldn't be too useful, and we would already have some log information from the `zpool history` log entry.

Reply:
Cool - thanks for the clarification. I think having the information in `zpool history` is sufficient